hazelcast/src/test/java/classloading/domain/IntMapLoader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package classloading.domain;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IntMapLoader implements com.hazelcast.core.MapLoader<Integer, Integer>, Serializable {
```

Comment:
Do we need to define this class as `Serializable`?

Reply:
It is necessary to observe the second mode of failure (failure to deserialize on the joining member) in the added test (when testing without the fix).